pynest/examples/pynest-microcircuit/helpers.py

```diff
@@ -0,0 +1,142 @@
+
+Hendrik Rothe, Hannah Bos, Sacha van Albada; May 2016
+'''
+
+import numpy as np
+from network_params import *
+from stimulus_params import *
+
+
+def get_weight(PSP_val):
+    """This function computes the weight which elicits a change in the membrane
+    potential of size PSP_val.
+
+    Arguements:
+    PSP_val: evoked postsynaptic potential
+
+    Returns: weight value
+
+    """
+    C_m = net_dict['n_params']['C_m']
+    tau_m = net_dict['n_params']['tau_m']
+    tau_ex = net_dict['n_params']['tau_ex']
+    tau_in = net_dict['n_params']['tau_in']
+
+    PSC_e_over_PSP_e = (((C_m)**(-1) * tau_m * tau_ex / (tau_ex - tau_m) * (
+        (tau_m / tau_ex)**(- tau_m / (tau_m-tau_ex)) - (
+            tau_m/tau_ex)**(- tau_ex/(tau_m-tau_ex))))**(-1))
+    PSC_e = (PSC_e_over_PSP_e * PSP_val)
+    return PSC_e
+
+
+def get_total_number_of_synapses():
+    """This function returns the total number of synapses between
+    all populations.
+
+    The first index (rows) of the matrix is the target population
+    and the second (columns) the source population.
+    This function depends on the following parameters specified in net_dict:
+    number of populations, Number of Neurons (N_full),
+    connection probability (conn_probs), scaling factor (N_scaling)
+
+    Returns: Matrix with total number of synapses,
+        dimension: (len(populations), len(populations)).
+    """
+    N_full = net_dict['N_full']
+    conn_probs = net_dict['conn_probs']
+    scaling = net_dict['N_scaling']
+    prod = np.outer(N_full, N_full)
+    n_syn_temp = np.log(1. - conn_probs)/np.log((prod - 1.) / prod)
+    N_full_matrix = np.column_stack((N_full for i in range(len(N_full))))
+    # if the network is scaled the indegrees are calculated in the same
+    # fashion as in the original version of the circuit, which is
+    # written in sli
+    K = (((n_syn_temp * (
+        N_full_matrix * scaling).astype(int)) / N_full_matrix).astype(int))
+    return K
+
+
+def synapses_th_matrix():
+    """This function returns the total number of synapses,
+        between the thalamus and all target populations.
+
+    Parameters: length of populations, number of neurons (N_full), connection
+        probability (conn_probs), scaling factor (N_scaling)
+
+    Returns: Vector with total number of synapses.
+    """
+    scaling = net_dict['N_scaling']
+    N_full = net_dict['N_full']
+    conn_probs = stim_dict['conn_probs_th']
+    scaling = net_dict['N_scaling']
+    prod = (stim_dict['n_thal'] * net_dict['N_full']).astype(float)
+    n_syn_temp = np.log(1. - conn_probs)/np.log((prod - 1.)/prod)
+    K = (((n_syn_temp * (N_full * scaling).astype(int))/N_full).astype(int))
+    return K
+
+
+def adj_w_ext_to_K(K_full, K_scaling, w, w_from_PSP, DC):
+    """With this funtion the recurrent and external weights are adjusted
+    to the scaling of the indegrees. Extra DC input is added to
+    compensate the scaling and preserve the mean and variance of the input.
```

Comment:
- The documentation is not complete because the internal weights are adapted as well.
- The documentation will become clearer if you add "Extra DC input I_ext is added"